lib/shower/player.js

```diff
@@ -54,72 +36,34 @@ class Player {
     }
 
     last() {
-        this.go(this._shower.getSlidesCount() - 1);
+        this.go(this.slidesCount - 1);
         return this;
     }
 
     /**
      * Go to custom slide by index.
      *
-     * @param {number | Slide} index Slide index to activate.
-     * @returns {Player}
+     * @param {number} slide index to activate.
      */
     go(index) {
-        // If go by slide instance.
-        if (typeof index !== 'number') {
-            index = this._shower.getSlideIndex(index);
-        }
-
-        const slidesCount = this._shower.getSlidesCount();
-        let slide = this._currentSlide;
-
-        if (index !== this._currentSlideIndex && index < slidesCount && index >= 0) {
-            if (slide && slide.isActive()) {
-                slide.deactivate();
-            }
-
-            slide = this._shower.get(index);
-            if (!slide.isActive()) {
-                slide.activate();
-            }
-
-            this._currentSlide = slide;
-            this._currentSlideIndex = index;
-            this.events.emit('activate', { index, slide });
+        const slide = this._slides[index];
+        if (slide) {
+            slide.activate();
         }
 
         return this;
     }
 
-    /**
-     * @returns {Slide} Current active slide.
-     */
-    getCurrentSlide() {
-        return this._currentSlide;
-    }
-
-    /**
-     * @returns {Number} Current active slide index.
-     */
-    getCurrentSlideIndex() {
+    get currentSlideIndex() {
         return this._currentSlideIndex;
     }
 
-    _onSlideAdd(event) {
-        const slide = event.get('slide');
-        slide.events.on('activate', this._onSlideActivate, this);
+    get currentSlide() {
```

Comment:
Maybe we should call it `activeSlide` to be on par with CSS class?

Reply:
```js
const index = player.activeSlideIndex;
const slide = player.activeSlide;
slide.deactivate();
slide.activate();
slide.isActive; // true
```

And of course, `.active`. IMO, mixing `current` and `active` when calling the same thing complicates learning the API.

@pepelsbey, any objections to rename `current` => `active`?